kube_hunter/modules/discovery/hosts.py

```diff
@@ -55,12 +55,8 @@ class HostDiscoveryHelpers:
     def get_cloud(host):
         try:
             logging.debug("Checking whether the cluster is deployed on azure's cloud")
-            # azurespeed.com provide their API via HTTP only; the service can be queried with
-            # HTTPS, but doesn't show a proper certificate. Since no encryption is worse then
-            # any encryption, we go with the verify=false option for the time being. At least
-            # this prevents leaking internal IP addresses to passive eavesdropping.
-            # TODO: find a more secure service to detect cloud IPs
-            metadata = requests.get("https://www.azurespeed.com/api/region?ipOrUrl={ip}".format(ip=host), verify=False).text
+            # Leverage 3rd tool https://github.com/blrchen/AzureSpeed for Azure cloud ip detection
+            metadata = requests.get("https://api.azurespeed.com/api/region?ipOrUrl={ip}".format(ip=host), verify=False).text
```

Comment:
Remove `verify=False`
```suggestion
            metadata = requests.get(f"https://api.azurespeed.com/api/region?ipOrUrl={host}").text
```